zio-http/shared/src/main/scala/zio/http/Form.scala

```diff
@@ -160,11 +160,10 @@ final case class Form(formData: Chunk[FormField]) {
   }
 
   def toQueryParams: QueryParams =
-    formData.foldLeft(QueryParams.empty) {
-      case (acc, FormField.Text(k, v, _, _)) => acc.addQueryParam(k, v)
-      case (acc, FormField.Simple(k, v))     => acc.addQueryParam(k, v)
-      case (acc, _)                          => acc
-    }
+    QueryParams(formData.collect {
+      case FormField.Text(k, v, _, _) => (k, Chunk(v))
+      case FormField.Simple(k, v)     => (k, Chunk(v))
```

Comment:
Have you tried not using the `unapply`? 
```scala
    QueryParams(formData.collect {
      case t: FormField.Text   => (t.k, Chunk(t.v))
      case s: FormField.Simple => (s.k, Chunk(s.v))
    }: _*)
```

That's a change the @kyri-petrou did a lot in ZIO codebase to improve perfs

Reply:
You are right. Should be even better. I'll make the change an check the benchmark again